lib/auth/auth.go

```diff
@@ -5707,6 +5715,82 @@ func (a *Server) syncDynamicLabelsAlert(ctx context.Context) {
 	}
 }
 
+func (a *Server) cleanupNotifications(ctx context.Context) {
+	userNotifications, _, err := a.Cache.ListUserNotifications(ctx, 0, "")
```

Comment:
This read should be rate-limited in case the cache is unhealthy and it ends up hitting the real backend.  Reads don't need to be rate-limited as aggressively as writes, but we should still keep it at or under about 200 reads per second I think.  Consider reading the notifications page-by-page with a small page size and a rate limiter.